src/promptflow-evals/tests/evals/e2etests/test_metrics_upload.py

```diff
@@ -131,6 +131,7 @@ def test_log_artifact(self, setup_data, caplog, tmp_path):
         ev_run.log_artifact(tmp_path)
         self._assert_no_errors_for_module(caplog.records, EvalRun.__module__)
 
+    @pytest.mark.performance_target_test

```

Comment:
Just thinking if it's straightforward to track the test run results and latency for all the E2E tests. We should be able to get GitHub workspace and action metadata through environment variables. This way, 1) we don't have to monitor specific tests 2) we don't have to re-run the same tests again.

Reply:
We can monitor the latency of all tests, however, if we will add the new test, it will be hard to track the regressions, as we will get additional lags, plus, we may want to have the additional knob to switch some test to "live" mode. May be as an alternative we can have one test group for regression tests, so that we will need to deliberately add tests there?

Reply:
Added tag for the performance test.

Reply:
Our test result report already includes the latency for all tests.

Check out this pipeline run: [Pipeline Run](https://github.com/microsoft/promptflow/actions/runs/9783328823). 

Go to "Artifacts" and download one of the test reports. After unzipping, open test-results.xml. You will find the test names along with their durations.

![image](https://github.com/microsoft/promptflow/assets/11171669/8cdffcc0-f7bd-4022-adeb-cbbfc31cf343)

Wonder if it's possible to convert this XML to a JSON payload and send it to AppInsights?

Reply:
Added code which parses JUnit XML and logs all test times to the application insights.